tools/c7n_azure/c7n_azure/session.py

```diff
@@ -100,6 +106,17 @@ def _initialize_session(self):
             self.tenant_id = os.environ[constants.ENV_TENANT_ID]
             self.log.info("Creating session with Service Principal Authentication")
 
+        elif all(k in os.environ for k in msi_auth_variables):
+            # MSI authentication
+            if constants.ENV_CLIENT_ID in os.environ:
+                self.credentials = MSIAuthentication(
+                    client_id=os.environ[constants.ENV_CLIENT_ID],
+                    resource=self.resource_namespace)
+            else:
+                self.credentials = MSIAuthentication(resource=self.resource_namespace)
```

Comment:
nit: I think this should also work (just one-liner)
self.credentials = MSIAuthentication(resource=self.resource_namespace, client_id=os.environ.get(constants.ENV_CLIENT_ID, None))

Reply:
When I did it this way I knew you'd wonder why :)  

The MSIAuthentication class doesn't check for parameters being null, but actually checks for the existence of keys in the argument list so I thought it better to not include it.
https://github.com/Azure/msrestazure-for-python/blob/master/msrestazure/azure_active_directory.py#L564

Reply:
I think you could do:
```
kwargs = {"resource":  self.resource_namespace}
if constants.ENV_CLIENT_ID in os.environ:
  kwargs['client_id'] = os.environ[constants.ENV_CLIENT_ID]

self.credentials = MSIAuthentication(**kwargs)
```

Reply:
For sure you could do that :)